DLCS.Repository.Tests/Assets/ThumbReorganiserTests.cs

```diff
@@ -278,5 +278,28 @@ public async Task EnsureNewLayout_AllowsConcurrentAttempts_ForDifferentKey()
             A.CallTo(() => assetRepository.GetAsset(A<string>._))
                 .MustHaveHappened(3, Times.Exactly);
         }
+
+        [Fact]
+        public async Task EnsureNewLayout_AssetNotFound()
+        {
+
+            // Arrange
+            var rootKey = new ObjectInBucket { Bucket = "the-bucket", Key = "2/1/doesnotexit/" };
+
+            Asset returnvalue = null;
+            A.CallTo(() => assetRepository.GetAsset(rootKey.Key.TrimEnd('/')))
+                     .Returns(returnvalue);
+            A.CallTo(() => bucketReader.GetMatchingKeys(rootKey))
+                    .Returns(new string[0]);

```

Comment:
Different issue but I _think_ we should bail from `EnsureNewLayout` if no matching keys are found as Thumbs can't create any new thumbs, it just shuffles them around.